src/button.js

```diff
@@ -4,39 +4,51 @@ import { makeStyles } from '@material-ui/core/styles'
 import MUIButton from '@material-ui/core/Button'
 import clsx from 'clsx'
 
-const useStyles = makeStyles((theme) => {
-  return {
-    label: {
-      color: 'rgba(0, 0, 0, 0.0) !important',
-    },
-    noSpacing: {
-      padding: theme.spacing(0),
-    },
-  }
-})
+const colorPalette = ['primary', 'secondary', 'red', 'green', 'teal', 'orange', 'yellow', 'purple', 'pink', 'brown']
+
+const useStyles = color => {
+  const selectColor = colorPalette.includes(color) ? color : 'primary'
+
+  const selectStyle = makeStyles((theme) => {
+    const customColor = theme.overrides.MuiButton(selectColor)
+
+    return {
+      containedPrimary: customColor.containedPrimary,
+      outlinedPrimary: customColor.outlinedPrimary,
+      textPrimary: customColor.textPrimary,
+      label: {
+        color: 'rgba(0, 0, 0, 0.0) !important',
+      },
+      noSpacing: {
+        padding: theme.spacing(0),
+      },
+    }
+  })
+
+  return selectStyle()
+}
```

Comment:
Alternatively, this part can be shorten with currying as: 

```jsx
const useStyles = (color) => makeStyles((theme) => {
  const selectColor = colorPalette.includes(color) ? color : 'primary'
  const customColor = theme.overrides.MuiButton(selectColor)
  
return {
    containedPrimary: customColor.containedPrimary,
    outlinedPrimary: customColor.outlinedPrimary,
    textPrimary: customColor.textPrimary,
    label: {
      color: 'rgba(0, 0, 0, 0.0) !important',
    },
    noSpacing: {
      padding: theme.spacing(0),
    },
  }
})
(...)
const classes = useStyles(color)()
```

Reply:
I was looking for a solution like that noice